### PR TITLE
Robustify production testing a little

### PR DESCRIPTION
**Pull request by MTCam**
- Replaces the old scripts with `production-install.sh` `production-driver-install.sh`, and `production-testing-env.sh`
- Developers only ever need to set up `production-testing-env.sh` which only sets variables
- Main never uses customizations - it only ever tests y1-production and nozzle driver without changes
- Supports specifying production branch outright
- Supports specifying production driver branch outright
- Add a gob of documentation to explain the settings/vars
- Makes explicit the defaulting of the owners to the local repo

### CODE REVIEW

**Review by MTCam — COMMENTED** [review]
Heh, this is a dumb change.  Not sure what I thought this would do.

**Comment by MTCam** [comment]
Ok - this now fixes the Y1 testing issues and cannot be messed up from merges.

File: .ci-support/install.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 if [ "$(uname)" = "Darwin" ]; then
 PLATFORM=MacOSX
 brew update
-brew upgrade
+# brew upgrade
```
Comment (thomasgibson) [inline .ci-support/install.sh:4]: Any reason to leave this here? We could also leave a short note explaining why this is commented out

**Review by thomasgibson — APPROVED** [review]
LGTM! Just a minor comment/question.

File: .ci-support/install.sh (hunk 1 above)
Comment (MTCam) [reply .ci-support/install.sh:4]: I overlooked/forgot it, I say we should eliminate with prejudice.

**Pull request merged**